app.py

```diff
@@ -277,179 +255,178 @@ def parse_repository_file_response(file_response: dict[str, Any]) -> str:
     blueprint_id = "bitbucketPullrequest"
 
     for pr in pullrequest_data:
         entity = {
-            "identifier": str(pr["id"]),
-            "title": pr["title"],
+            "identifier": str(pr.get("id")),
+            "title": pr.get("title"),
             "properties": {
-                "created_on": convert_to_datetime(pr["createdDate"]),
-                "updated_on": convert_to_datetime(pr["updatedDate"]),
-                "merge_commit": pr["fromRef"]["latestCommit"],
+                "created_on": convert_to_datetime(pr.get("createdDate")),
+                "updated_on": convert_to_datetime(pr.get("updatedDate")),
+                "merge_commit": pr.get("fromRef", {}).get("latestCommit"),
                 "description": pr.get("description"),
-                "state": pr["state"],
-                "owner": pr["author"]["user"]["displayName"],
-                "link": pr["links"]["self"][0]["href"],
-                "destination": pr["toRef"]["displayId"],
+                "state": pr.get("state"),
+                "owner": pr.get("author", {}).get("user", {}).get("displayName"),
+                "link": pr.get("links", {}).get("self", [{}])[0].get("href"),
+                "destination": pr.get("toRef", {}).get("displayId"),
                 "reviewers": [
-                    user["user"]["displayName"] for user in pr.get("reviewers", [])
+                    user.get("user", {}).get("displayName") for user in pr.get("reviewers", [])
                 ],
-                "source": pr["fromRef"]["displayId"],
+                "source": pr.get("fromRef", {}).get("displayId"),
             },
             "relations": {
                 "repository": pr["toRef"]["repository"]["slug"],
-                "participants": [pr.get("author")["user"]["emailAddress"]]
-                + [user["user"]["emailAddress"] for user in pr.get("participants", [])],
+                "participants": [pr.get("author", {}).get("user", {}).get("emailAddress")]
+                                + [user.get("user", {}).get("emailAddress") for user in pr.get("participants", [])],
             },
         }
-        add_entity_to_port(blueprint_id=blueprint_id, entity_object=entity)
+        await add_entity_to_port(blueprint_id=blueprint_id, entity_object=entity)
 
-
-def get_repository_readme(project_key: str, repo_slug: str) -> str:
+async def get_repository_readme(project_key: str, repo_slug: str) -> str:
     file_path = f"projects/{project_key}/repos/{repo_slug}/browse/README.md"
     readme_content = ""
-    for readme_file_batch in get_paginated_resource(
-        path=file_path, page_size=500, full_response=True
+    async for readme_file_batch in get_paginated_resource(
+            path=file_path, page_size=500, full_response=True
     ):
         file_content = parse_repository_file_response(readme_file_batch)
         readme_content += file_content
     return readme_content
 
```

Comment:
There was an error where the app crashes because a repository does not have a README.md I didn't see how it is addressed in this PR

Reply:
It has been addressed, the order of handling the error in the get_paginated_resources is the problem with the previous implementation. I have fixed the order of the error handling in the get paginated resources to deal with the 404 differently and not re-raise it when done before moving on.